driver/bk4819.c

```diff
@@ -196,83 +195,79 @@ void BK4819_WriteRegister(bk4819_register_t Register, uint16_t Data)
 	GPIO_SetBit(&GPIOC->DATA, GPIOC_PIN_BK4819_SCN);
 	GPIO_ClearBit(&GPIOC->DATA, GPIOC_PIN_BK4819_SCL);
 
-	SYSTICK_DelayUs(1);
+	SYSTICK_DelayUs(1U);
 
 	GPIO_ClearBit(&GPIOC->DATA, GPIOC_PIN_BK4819_SCN);
 	BK4819_WriteU8(Register);
 
-	SYSTICK_DelayUs(1);
+	SYSTICK_DelayUs(1U);
 
 	BK4819_WriteU16(Data);
 
-	SYSTICK_DelayUs(1);
+	SYSTICK_DelayUs(1U);
 
 	GPIO_SetBit(&GPIOC->DATA, GPIOC_PIN_BK4819_SCN);
 
-	SYSTICK_DelayUs(1);
+	SYSTICK_DelayUs(1U);
 
 	GPIO_SetBit(&GPIOC->DATA, GPIOC_PIN_BK4819_SCL);
 	GPIO_SetBit(&GPIOC->DATA, GPIOC_PIN_BK4819_SDA);
 }
 
 void BK4819_WriteU8(uint8_t Data)
 {
-	unsigned int i;
-
 	GPIO_ClearBit(&GPIOC->DATA, GPIOC_PIN_BK4819_SCL);
-	for (i = 0; i < 8; i++)
+	for (int32_t i = 7; i >= 0; i--)
 	{
-		if ((Data & 0x80) == 0)
-			GPIO_ClearBit(&GPIOC->DATA, GPIOC_PIN_BK4819_SDA);
-		else
+		if ((Data & (1 << i)))
 			GPIO_SetBit(&GPIOC->DATA, GPIOC_PIN_BK4819_SDA);
+		else
+			GPIO_ClearBit(&GPIOC->DATA, GPIOC_PIN_BK4819_SDA);
 
-		SYSTICK_DelayUs(1);
-		GPIO_SetBit(&GPIOC->DATA, GPIOC_PIN_BK4819_SCL);
-		SYSTICK_DelayUs(1);
+		SYSTICK_DelayUs(1U);
 
-		Data <<= 1;
+		GPIO_SetBit(&GPIOC->DATA, GPIOC_PIN_BK4819_SCL);
+		SYSTICK_DelayUs(1U);
 
 		GPIO_ClearBit(&GPIOC->DATA, GPIOC_PIN_BK4819_SCL);
-		SYSTICK_DelayUs(1);
+		SYSTICK_DelayUs(1U);
 	}
 }
 
 void BK4819_WriteU16(uint16_t Data)
 {
-	unsigned int i;
-
 	GPIO_ClearBit(&GPIOC->DATA, GPIOC_PIN_BK4819_SCL);
-	for (i = 0; i < 16; i++)
+	for (int32_t i = 15; i >= 0; i--)
 	{
-		if ((Data & 0x8000) == 0)
-			GPIO_ClearBit(&GPIOC->DATA, GPIOC_PIN_BK4819_SDA);
-		else
+		if ((Data & (1 << i)))
 			GPIO_SetBit(&GPIOC->DATA, GPIOC_PIN_BK4819_SDA);
+		else
+			GPIO_ClearBit(&GPIOC->DATA, GPIOC_PIN_BK4819_SDA);
 
-		SYSTICK_DelayUs(1);
-		GPIO_SetBit(&GPIOC->DATA, GPIOC_PIN_BK4819_SCL);
+		SYSTICK_DelayUs(1U);
 
-		Data <<= 1;
+		GPIO_SetBit(&GPIOC->DATA, GPIOC_PIN_BK4819_SCL);
+		SYSTICK_DelayUs(1U);
 
-		SYSTICK_DelayUs(1);
 		GPIO_ClearBit(&GPIOC->DATA, GPIOC_PIN_BK4819_SCL);
-		SYSTICK_DelayUs(1);
+		SYSTICK_DelayUs(1U);
 	}
 }
 
-void BK4819_SetAGC(uint8_t Value)
+void BK4819_SetAGC(bool Enable)
 {
-	if (Value == 0)
+	if (Enable)
```

Comment:
It seems you changed a "if false" to a "if true".

Reply:
Nope, the if blocks are swapped, at the same time, just to avoid a 'if (!Enabled) { .. } lse { .. }